### DISCUSSION TITLE
Setup Mess Menu

### DISCUSSION THREAD

**Discussion by DesignrKnight** [dscnitrourkela/project-dates#88] [Ideas]
Hey, @SriramPatibanda this is the decision for now. Take  reference from the other discussion on Timetable for reference

**Comment by DesignrKnight** [comment]
![image](https://user-images.githubusercontent.com/27865704/106881903-1aace280-6704-11eb-8cb7-72ea2f1084e3.png)

**Reply by SriramPatibanda** [reply]
![WhatsApp Image 2021-01-29 at 12 12 01 AM](https://user-images.githubusercontent.com/56649197/107154923-8361bc80-699b-11eb-9cf8-d32d3b9c53a5.jpeg)

**Reply by DesignrKnight** [reply]
v1

```
{
  "messName": "K.R.Patel", (String)
  "mess": [
    {
      "Day": "Monday"(string),
      "meal": {
        "items": [
          {
            "name": "Upma"(String),
            "mealType": "Breakfast"(String),
            "itemType": "General item" (String)
          }
        ]
      }
    }
  ]
}
```

**Comment by DesignrKnight** [comment]
**Possible Queries**
- Fetch menu of the given mess(full, given day)
- Update/Add menu

**Comment by DesignrKnight** [comment]
Item categorisation
- Common item: like tea, rice, milk, dal
- set of items from which you can choose one set: like [veg, non-veg] [dosa, puri, bread]

**Reply by DesignrKnight** [reply]
```
commonItem: string
selectableItems: [{
item: String
isVeg: Boolean
}]
```
Query: isVeg: Boolean vs itemType: String/Integer

**Reply by DesignrKnight** [reply]
**Mess Menu Sample**
- chicken curry and chole and aalo
- paneer curry and chole and aalo

```
commonItem: "Roti - Rice - Rasam"
selectableItems: [{
item: "chicken curry - chole - aalo"
isVeg: false
}
{
item: "paneer curry - chole - aalo"
isVeg: true
}
]
```

**Reply by DesignrKnight** [reply]
Should chole aalo be pushed to common item

**Reply by DesignrKnight** [reply]
Let us call the above object as
meal: {
commonItem, selectableItem}

**Reply by DesignrKnight** [reply]
```
[{
breakfast: meal
lunch: meal
snacks: meal
dinner: meal
}]
```
n=0 to 6


**OR**
```
{
sun: {
breakfast: meal
lunch: meal
snacks: meal
dinner: meal
}
mon: {
breakfast: meal
lunch: meal
snacks: meal
dinner: meal
}
}
```

Vote
🚀: First
🎉: Second

**Reply by DesignrKnight** [reply]
@SriramPatibanda if we make `commonItems` into an array-like `selectableItems`, it would works
@HarishTeens and I were discussing and it seems to work
